packages/kbn-es/src/utils/docker.ts

```diff
@@ -621,7 +629,25 @@ export async function runServerlessCluster(log: ToolingLog, options: ServerlessO
 
   if (options.waitForReady) {
     log.info('Waiting until ES is ready to serve requests...');
-    await waitUntilClusterReady();
+
+    const esNodeUrl = `${options.ssl ? 'https' : 'http'}://${portCmd[1].substring(
+      0,
+      portCmd[1].lastIndexOf(':')
+    )}`;
+
+    await waitUntilClusterReady({
+      node: esNodeUrl,
+      ...(options.ssl
+        ? {
+            auth: { bearer: kibanaDevServiceAccount.token },
+            tls: {
```

Comment:
Out of curiosity can we try with the following tls settings?

```
[Fs.readFileSync(CA_CERT_PATH)]
tls: {
  key: fs.readFileSync(KBN_KEY_PATH),
  cert: fs.readFileSync(KBN_CERT_PATH),
  requestCert: true,
  ca: [ fs.readFileSync(CA_CERT_PATH) ],
}
```

I'm not sure about `key` and `cert` options but lets try with and without them (they can be required from @kbn/dev-utils).

Reply:
Update, the following will do it:

```
await waitUntilClusterReady({
      node: esNodeUrl,
      ...(options.ssl
        ? {
          auth: { bearer: kibanaDevServiceAccount.token },
          tls: {
            ca: [fs.readFileSync(CA_CERT_PATH)],
            // NOTE: Even though we've added ca into the tls options, we are using 127.0.0.1 instead of localhost
            // for the ip which is not validated. As such we are getting the error
            // Hostname/IP does not match certificate's altnames: IP: 127.0.0.1 is not in the cert's list:
            // To work around that we are overriding the function checkServerIdentity too
            checkServerIdentity: () => { return undefined },
          },
        }
        : {}),
    });
```